topology/src/test/java/io/camunda/zeebe/topology/api/TopologyManagementRequestsHandlerTest.java

```diff
@@ -0,0 +1,118 @@
+import io.camunda.zeebe.topology.util.RoundRobinPartitionDistributor;
+import io.camunda.zeebe.topology.util.TopologyUtil;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+import java.util.stream.Stream;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+
+class TopologyManagementRequestsHandlerTest {
+
+  private final RecordingChangeCoordinator coordinator = new RecordingChangeCoordinator();
+  private final TopologyManagementRequestsHandler handler =
+      new TopologyManagementRequestsHandler(coordinator, new TestConcurrencyControl());
+
+  @ParameterizedTest
+  @MethodSource("provideReassignmentParameters")
+  void shouldReassignPartitionsRoundRobin(
+      final int partitionCount,
+      final int replicationFactor,
+      final int oldClusterSize,
+      final int newClusterSize) {
+    // given
+    final var expectedNewDistribution =
+        new RoundRobinPartitionDistributor()
+            .distributePartitions(
+                getClusterMembers(newClusterSize),
+                getSortedPartitionIds(partitionCount),
+                replicationFactor);
+
+    final var oldDistribution =
+        new RoundRobinPartitionDistributor()
+            .distributePartitions(
+                getClusterMembers(oldClusterSize),
+                getSortedPartitionIds(partitionCount),
+                replicationFactor);
+    var oldClusterTopology = TopologyUtil.getClusterTopologyFrom(oldDistribution);
+    for (int i = oldClusterSize; i < newClusterSize; i++) {
+      oldClusterTopology =
+          oldClusterTopology.addMember(
+              MemberId.from(Integer.toString(i)), MemberState.initializeAsActive(Map.of()));
+    }
+    coordinator.setCurrentTopology(oldClusterTopology);
+
+    // when
+    handler
+        .reassignPartitions(new ReassignPartitionsRequest(getClusterMembers(newClusterSize)))
+        .join();
+
+    // apply operations to generate new topology
+    final var topologyChangeSimulator =
+        new TopologyChangeAppliersImpl(
+            new NoopPartitionChangeExecutor(), new NoopTopologyMembershipChangeExecutor());
+    var newTopology = oldClusterTopology.startTopologyChange(coordinator.getLastAppliedOperation());
+
+    while (newTopology.hasPendingChanges()) {
+      final var operation = newTopology.changes().pendingOperations().get(0);
+      final var applier = topologyChangeSimulator.getApplier(operation);
+      newTopology = newTopology.updateMember(operation.memberId(), applier.init(newTopology).get());
+      newTopology = newTopology.advanceTopologyChange(operation.memberId(), applier.apply().join());
+    }
+
+    // then
+    final var newDistribution = TopologyUtil.getPartitionDistributionFrom(newTopology, "temp");
+    assertThat(newDistribution).isEqualTo(expectedNewDistribution);
```

Comment:
:wrench: `PartitionMetadata` implements `equals` in terms of partition id and members but ignores the priority and other properties. Can't we just compare `newTopology` with `oldClusterTopology` using `ClusterTopologyAssert`?

Reply:
Oh.. good catch. Will change to verify using topology.

Reply:
I updated `PartitionMetadata#equals` to compare all properties in it including priority. This would be address this case. There is no reason to not include them in the equals.